services/Notifications.ts

```diff
@@ -0,0 +1,128 @@
+    if (existingStatus !== "granted") {
+      const { status } = await ExpoNotifications.requestPermissionsAsync();
+      finalStatus = status;
+    }
+    if (finalStatus !== "granted") {
+      errorToast(
+        new Error(
+          "Permission not granted to get push token for push notification",
+        ),
+      );
+      return;
+    }
+    const projectId =
+      Constants?.expoConfig?.extra?.eas?.projectId ??
+      Constants?.easConfig?.projectId;
+    if (!projectId) {
+      errorToast(new Error("Project ID not found"));
+    }
+    try {
+      const pushTokenString = (
+        await ExpoNotifications.getExpoPushTokenAsync({
+          projectId,
+        })
+      ).data;
+
+      // FIXME: This would trigger the same notification
+      // per wallet if they belong to the same node
+      const wallets = useAppStore.getState().wallets;
+      let groupDefaults;
+      if (Platform.OS === "ios") {
+        groupDefaults = new UserDefaults(SUITE_NAME);
+      }
+
+      for (const wallet of wallets) {
+        const nwcUrl = wallet.nostrWalletConnectUrl;
+        if (!nwcUrl) {
+          continue;
+        }
+        const nwcClient = new nwc.NWCClient({
+          nostrWalletConnectUrl: wallet.nostrWalletConnectUrl,
+        });
+
+        const body = {
+          pushToken: pushTokenString,
+          relayUrl: nwcClient.relayUrl,
+          connectionPubkey: nwcClient.publicKey,
+          walletPubkey: nwcClient.walletPubkey,
+        };
+
+        try {
+          const response = await fetch(
+            `${NOSTR_API_URL}/nip47/notifications/go`,
+            {
+              method: "POST",
+              headers: {
+                Accept: "application/json",
+                "Accept-encoding": "gzip, deflate",
+                "Content-Type": "application/json",
+              },
+              body: JSON.stringify(body),
+            },
+          );
+
+          if (response.ok) {
+            const responseData = await response.json();
+            useAppStore.getState().updateWallet(
+              {
+                pushId: responseData.subscriptionId,
+              },
+              wallet.nostrWalletConnectUrl,
+            );
+            // TODO: Send a DELETE call to nostr api on deleting the wallet
+          } else {
+            throw new Error(`Error: ${response.status} ${response.statusText}`);
+          }
+
+          if (groupDefaults) {
+            let wallets = (await groupDefaults.get("wallets")) || {};
+            wallets[nwcClient.publicKey] = {
+              name: wallet.name,
+              sharedSecret: computeSharedSecret(
+                nwcClient.walletPubkey,
+                nwcClient.secret ?? "",
+              ),
+            };
+            groupDefaults.set("wallets", wallets);
+          }
+        } catch (error) {
+          errorToast(error);
+        }
+      }
+
+      return;
+    } catch (error) {
+      errorToast(error);
+    }
+  } else {
```

Comment:
rather than a huge if content with an else right at the bottom, could you do the check at the top and exit the function early?